internal/pkg/agent/cmd/diagnostics.go

```diff
@@ -156,6 +163,72 @@ func createZip(fileName string, agentDiag []client.DiagnosticFileResult, unitDia
 	return closeHandlers(nil, zw, f)
 }
 
+func writeRedacted(streams *cli.IOStreams, fullFilePath string, fr client.DiagnosticFileResult, w io.Writer) error {
+	out := &fr.Content
+
+	// Should we support json too?
+	if fr.ContentType == "application/yaml" {
+		unmarshalled := map[string]interface{}{}
+		err := yaml.Unmarshal(fr.Content, &unmarshalled)
+
+		if err != nil {
+			// Best effort, output a warning but still include the file
+			fmt.Fprintf(streams.Err, "[warning] Could not redact %s due to unmarshalling error: %s\n", fullFilePath, err)
+		} else {
+			redacted, err := yaml.Marshal(redactMap(unmarshalled))
+			if err != nil {
+				// Best effort, output a warning but still include the file
+				fmt.Fprintf(streams.Err, "[warning] Could not redact %s due to marshalling error: %s\n", fullFilePath, err)
+			}
+	
+			out = &redacted
```

Comment:
This needs to be an in else to avoid modifying when there is a marshalling error